core/kernel.go

```diff
@@ -228,38 +243,26 @@ func (kern *Kernel) LoadTransactor() (err error) {
 	nameRegState := kern.State
 	kern.Service = rpc.NewService(accountState, nameRegState, kern.Blockchain, kern.State, nodeView, kern.Logger)
 
+	// If no Tendermint node is configured load a no-consensus Transactor that bypasses Tendermint
 	if kern.Node == nil {
-		checkTx := func(tx tmTypes.Tx, cb func(*abciTypes.Response)) error {
-			exec := abci.NewTxExecutor(kern.nodeInfo, kern.exeChecker, kern.exeCommitter, kern.txCodec, kern.Logger.WithScope("CheckTx"))
-			ctr := exec.CheckTx(tx)
-			dtr := exec.DeliverTx(tx)
-			appHash, err := kern.exeCommitter.Commit(nil)
-			if err != nil {
-				return err
-			}
-
-			if err := kern.Blockchain.CommitBlock(time.Now(), nil, appHash); err != nil {
-				return err
-			}
-
-			cb(abciTypes.ToResponseCheckTx(ctr))
-			cb(abciTypes.ToResponseDeliverTx(dtr))
-			cb(abciTypes.ToResponseCommit(abciTypes.ResponseCommit{
-				Data: appHash,
-			}))
-
-			return nil
-		}
-
-		kern.Transactor = execution.NewTransactor(kern.Blockchain, kern.Emitter,
-			execution.NewAccounts(kern.exeChecker, kern.keyClient, AccountsRingMutexCount),
-			checkTx, kern.txCodec, kern.Logger)
-	} else {
-		kern.Blockchain.SetBlockStore(bcm.NewBlockStore(nodeView.BlockStore()))
-		kern.Transactor = execution.NewTransactor(kern.Blockchain, kern.Emitter,
-			execution.NewAccounts(kern.exeChecker, kern.keyClient, AccountsRingMutexCount),
-			kern.Node.MempoolReactor().Mempool.CheckTx, kern.txCodec, kern.Logger)
+		// Provide execution accounts against backend state since we will commit immediately
+		accounts := execution.NewAccounts(kern.checker, kern.keyClient, AccountsRingMutexCount)
+		// Elide consensus and use a CheckTx function that immediately commits any valid transaction
+		checkTx := abci.TxCommitter(kern.checker, kern.committer, kern.Blockchain, kern.txCodec,
+			time.Duration(kern.timeoutFactor*float64(time.Second)))
+		kern.Transactor = execution.NewTransactor(kern.Blockchain, kern.Emitter, accounts, checkTx, kern.txCodec,
+			kern.Logger)
+		return nil
 	}
+
+	// Use the Tendermint blockstore
+	kern.Blockchain.SetBlockStore(bcm.NewBlockStore(nodeView.BlockStore()))
+	// Provide execution accounts against checker state so that we can assign sequence numbers
+	accounts := execution.NewAccounts(kern.checker, kern.keyClient, AccountsRingMutexCount)
+	// Pass transactions to Tendermint's CheckTx function for broadcast and consensus
+	checkTx := kern.Node.MempoolReactor().Mempool.CheckTx
+	kern.Transactor = execution.NewTransactor(kern.Blockchain, kern.Emitter, accounts, checkTx, kern.txCodec,
```

Comment:
If you want to simplify this even further I guess you only need to set `checkTx` and possibly `kern.Blockchain` if in consensus mode.